### PR TITLE
feat: header styles

### PR DESCRIPTION
**Pull request by alva-fresh**
## Description

Added styles for Semantic UI `Header` component based on _Typography_ styles

## Screenshots

Screenshot 2023-05-18 at 2 29 47 PM

## Notes

Semantic Header: https://react.semantic-ui.com/elements/header/
https://dev.azure.com/EY-GA/Global%20Atlantic%20DevOps/_workitems/edit/23129

### CODE REVIEW

File: src/styles/variables.css [hunk 1]
```diff
@@ -56,6 +56,44 @@
...
   --gotham: 'Gotham', Arial, Calibri, sans-serif;
 
+  /* font size */
+
+  --font-size: 1rem;
+  --line-height: 1.5rem;
+
+  --display1: 3.75rem;
+  --display2: 3.5rem;
+  --h1: 2.25rem;
```
Comment (michael-lynch) [inline src/styles/variables.css]: Let's group the font size and line height variables together, as we generally want to know what they are relative to one another. Let's also use `font-size` and `line-height` to be more explicit.

```
--h1-font-size: 1rem;
--h1-line-height: 3.375rem;

--h2-font-size: 2rem;
--h2-line-height: 3rem;
```

**Review by michael-lynch — CHANGES_REQUESTED** [review]

File: src/styles/header.css [hunk 2]
```diff
@@ -1,3 +1,71 @@
...
+h5.ui.header {
+  font-size: var(--h5);
+  line-height: var(--h5-height);
+  font-weight: var(--black-weight);
+}
+
+h6.ui.header {
+  font-size: var(--subheading);
+  line-height: var(--subheading-height);
+  font-weight: var(--book-weight);
```
Comment (michael-lynch) [inline src/styles/header.css]: @Dart-EY This is where the font weight variables get a little tricky. It turns out that not all headings use the same weight, so `--weight-heading` won't be sufficient. I guess we could introduce `--weight-subheading` but that's a little ambiguous. It's probably easier if we just write `400` directly.

File: src/styles/variables.css [hunk 3]
```diff
@@ -56,6 +56,50 @@
   /* font */
   --gotham: 'Gotham', Arial, Calibri, sans-serif;
 
+  /* font size */
+
+  --font-size: 1rem;
+  --line-height: 1.5rem;
+
+  --display1-font-size: 3.75rem;
+  --display1-line-height: 5.625rem;
```
Comment (michael-lynch) [inline src/styles/variables.css]: Let's use the Semantic nomenclature.

"subheading" = "h6"
"display1" = "huge"
"display2" = "large"
"p1" = "medium"
"p2" = "small"
"p3" = "tiny"

File: src/styles/header.css (hunk 2 above)
Comment (Dart-EY) [reply src/styles/header.css]: Got it. I will make the changes to the input PR.

File: src/components/pages/Theme/Theme.jsx [hunk 4]
```diff
@@ -233,7 +233,7 @@ export const Theme = () => {
           <Header as='h3'>Heading 3</Header>
           <Header as='h4'>Heading 4</Header>
           <Header as='h5'>Heading 5</Header>
-          <Header as='h5'>Heading 5</Header>
+          <Header as='h6'>Heading 6</Header>
           <Header size='medium'>Medium</Header>
           <Header size='small'>Small</Header>
           <Header size='tiny'>Tiny</Header>
```
Comment (Dart-EY) [inline src/components/pages/Theme/Theme.jsx:256]: I was taking a look at this and I thought a couple of things about the `Header` component. We don't know if the variants `small`, `medium`, `tiny`, huge` and `large` are being used as we are thinking. 

For instance, we are setting the `huge` variant as it was the `display1` that is in the Figma designs. But what if Rahul is using the `huge` in a different way, our stylesheet would affect it.

File: src/styles/variables.css [hunk 5]
```diff
@@ -56,6 +56,44 @@
...
+  --h5-height: 1.875rem;
+
+  --subheading-height: 1.6rem;
+  --p1-height: 1.5rem;
+  --p2-height: 1.3rem;
+  --p3-height: 1.125rem;
+
+  /* font weights */
+
+  --black-weight: 900;
```
Comment (Dart-EY) [inline src/styles/variables.css]: I believe we are not going to use the variables anymore, right? Mike said we should use the values instead. At least that's what I understood from his comments in the Input PR and his comment above.

Comment (alva-fresh) [reply src/styles/variables.css]: It might make sense to keep it for Typography

Comment (michael-lynch) [reply src/styles/variables.css]: Yeah, let's just use the `400`, `700` etc. values directly.

**Review by michael-lynch — COMMENTED** [review]
Once the conflicts are resolved and the weights have been updated to use direct values, I think we can merge this.

**Review by michael-lynch — APPROVED** [review]

**Pull request merged**